plugins/repository-azure/src/main/java/org/elasticsearch/plugin/repository/azure/AzureRepositoryPlugin.java

```diff
@@ -53,7 +54,18 @@ protected AzureStorageService createStorageService(Settings settings) {
 
     @Override
     public List<Setting<?>> getSettings() {
-        return Collections.singletonList(AzureStorageService.Storage.STORAGE_ACCOUNTS);
+        return Arrays.asList(
+            AzureStorageService.Storage.STORAGE_ACCOUNTS,
+            // This is strange that deprecated group settings have never been registered before.
+            // How this worked? I was expecting something like:
+            // AzureStorageSettings.DEPRECATED_ACCOUNT_SETTING,
```

Comment:
These should definitely be registered. Is there some other registration of a prefix that is masking validation of all of these settings?